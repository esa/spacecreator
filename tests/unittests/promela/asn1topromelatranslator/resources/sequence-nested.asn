Sequence-nested DEFINITIONS ::=
BEGIN

Inner-sequence ::= SEQUENCE {
    write-address INTEGER (0 .. 10)
}

Outer-sequence ::= SEQUENCE {
   mywritevalue Inner-sequence
}

END
