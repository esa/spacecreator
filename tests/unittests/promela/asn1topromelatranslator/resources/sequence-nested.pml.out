inline Inner_sequence_write_address_generate_value(Inner_sequence_write_address_gv)
{
    if
    ::  true;
        Inner_sequence_write_address_gv = 0;
    ::  true;
        Inner_sequence_write_address_gv = 1;
    ::  true;
        Inner_sequence_write_address_gv = 2;
    ::  true;
        Inner_sequence_write_address_gv = 3;
    ::  true;
        Inner_sequence_write_address_gv = 4;
    ::  true;
        Inner_sequence_write_address_gv = 5;
    ::  true;
        Inner_sequence_write_address_gv = 6;
    ::  true;
        Inner_sequence_write_address_gv = 7;
    ::  true;
        Inner_sequence_write_address_gv = 8;
    ::  true;
        Inner_sequence_write_address_gv = 9;
    ::  true;
        Inner_sequence_write_address_gv = 10;
    fi;
}
inline Inner_sequence_generate_value(Inner_sequence_gv)
{
    Inner_sequence_write_address_generate_value(Inner_sequence_gv.write_address);
}
inline Outer_sequence_generate_value(Outer_sequence_gv)
{
    Inner_sequence_generate_value(Outer_sequence_gv.mywritevalue);
}
